cognite_toolkit/_cdf_tk/commands/pull.py

```diff
@@ -635,45 +712,66 @@ def _to_write_content(
             T_ID, T_WriteClass, T_WritableCogniteResource, T_CogniteResourceList, T_WritableCogniteResourceList
         ],
     ) -> tuple[str, dict[Path, str]]:
-        # 1. Replace all variables
+        # 1. Replace all variables with placeholders
         # 2. Load source and keep the comments
-        # 3. Update all items with the to_write values..
-        # 4. Dump the yaml
-        # 5. Add the variables back
+        # 3. Update the to_write dict with the placeholders
+        # 4. Dump the yaml with the placeholders
+        # 5. Replace the placeholders with the variables
+        # 6. Add the comments back
```

Comment:
Realized that the current procedure was not robust, to changed it to this new procedure.